### DISCUSSION TITLE
Unable to connect to plugin Html Report inside Docker

### DISCUSSION THREAD

**Discussion by Gaugehej93** [getgauge/gauge#2241] [General]
# Versions
NPM version: 8.6.0
Gauge version: 1.4.3
Taiko version: latest
Node version: v18.0.0

# Dockerfile
```
# This image uses the official node base image.
FROM node

# The Taiko installation downloads and installs the chromium required to run the tests. 
# However, we need the chromium dependencies installed in the environment. These days, most 
# Dockerfiles just install chrome to get the dependencies.
RUN apt-get update \
     && apt-get install -y wget gnupg ca-certificates \
     && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
     && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
     && apt-get update \
     && apt-get install -y google-chrome-stable
 
# Set a custom npm install location so that Gauge, Taiko and dependencies can be 
# installed without root privileges
ENV NPM_CONFIG_PREFIX=/home/gauge/.npm-packages
ENV PATH="${NPM_CONFIG_PREFIX}/bin:/home/gauge:${PATH}"
# Add the Taiko browser arguments
ENV TAIKO_BROWSER_ARGS=--no-sandbox,--start-maximized,--disable-dev-shm-usage
ENV headless_chrome=true
ENV TAIKO_SKIP_DOCUMENTATION=true

# Uncomment the lines below to use chrome bundled with this image
#ENV TAIKO_SKIP_CHROMIUM_DOWNLOAD=true
#ENV TAIKO_BROWSER_PATH=/usr/bin/google-chrome
 
# Set working directory
WORKDIR /gauge

# Copy the local working folder
COPY . .

# Create an unprivileged user to run Taiko tests
RUN groupadd -r gauge && useradd -r -g gauge -G audio,video gauge && \
   mkdir -p /home/gauge/.npm-packages/lib && \
   chown -R gauge:gauge /home/gauge /gauge
 
USER gauge

RUN curl -SsL https://downloads.gauge.org/stable | sh -s -- --location=/home/gauge
RUN gauge install html-report --version 4.0.10
RUN npm install

RUN gauge --version
RUN node --version

# Default command on running the image
ENTRYPOINT ["gauge", "run", "specs"]

```

# Error
Running the image with the following command
```bash
docker run  --rm -it -v ${PWD}/reports:/gauge/reports gauge-taiko
```
Executes the tests sucessfully but fails in generating the html report

> Unable to connect to plugin Html Report 4.0.10. rpc error: code = Unavailable desc = error reading from server: EOF
> Error occurred while waiting for plugin process to finish.
> Error : exit status 1
> Failed to kill plugin Html Report 4.0.10. os: process already finished

**Comment by zabil** [comment]
> RUN gauge install html-report --version 4.0.10

Can you try changing this to

```
RUN gauge install html-report
```

**Reply by pjvilloud** [reply]
Hello ! I'm having the same issue with html-report when using Docker (with the stock Dockerfile generated by gauge). Always having it on my Jenkins, and kind of randomly (well probably not randomly but I did not manage to find what causes this issue...) having it on my local computer. The versions are : 

Gauge version: 1.4.3
Commit Hash: f98dd40

Plugins
-------
html-report (4.1.4)
js (2.4.0)
screenshot (0.1.0)

Any idea ?

**Reply by sriv** [reply]
Hi @pjvilloud - do you see any hints in the logs?

**Reply by pjvilloud** [reply]
Hi thanks for your answer @sriv I only see that : 
`Unable to connect to plugin Html Report 4.1.4. rpc error: code = Unavailable desc = error reading from server: EOF`
The tests are running fine

**Reply by sriv** [reply]
that isn't very helpful. I'll try to see if I can replicate this in my local.

**Reply by pjvilloud** [reply]
Sorry :( I don't know if I can turn on some verbose setting ? I can share my Jenkinsfile : 
```
pipeline {
    agent {
        label 'main'
    }
    tools {
        'org.jenkinsci.plugins.docker.commons.tools.DockerTool' 'Docker_20'
    }
    stages {
        stage('Build') {
            steps {
                sh "docker build -t gauge-taiko ."
                sh "docker run  --rm -i -v ${WORKSPACE}/reports:/gauge/reports gauge-taiko"
            }
        }
    }
}

**Reply by sriv** [reply]
Sorry, I wasn't implying that your report isn't helpful, rather was blaming the logs (and us indirectly, since we didn't capture enough details).

**Reply by pjvilloud** [reply]
FYI, the error message disappeared in my last builds. No clue why :) 

Now the html report directory is not exported into the Jenkins workspace, but it's probably my configuration that needs an adjustement.

Thanks anyway and keep up the good work !
